flocker/control/_diffing.py

```diff
@@ -76,7 +79,9 @@ class _Set(PClass):
     value = field()
 
     def apply(self, obj):
-        return obj.transform(self.path, self.value)
+        return obj.transform(
+            self.path[:-1], lambda o: o.set(self.path[-1], self.value)
```

Comment:
It'd be nice to have docs on the interface expected of `obj` and `self.path`.  They look pyrsistent-y but this change seems kind of subtle so I'm not sure.